File: src/main/java/tv/comnata/comnata/controllers/MainController.java
package tv.comnata.comnata.controllers;

import org.springframework.web.bind.annotation.*;
import tv.comnata.comnata.pojo.Cat;

import javax.servlet.http.HttpServletRequest;

@RestController
@RequestMapping("/main")
public class MainController {
    @GetMapping("/{id}")
    @ResponseBody
    public Cat index(HttpServletRequest request, @PathVariable int id) {
        return new Cat(id, "Yo!", 18);
    }
}

File: main-service/src/main/kotlin/tv/comnata/mainservice/controllers/MainController.kt
package tv.comnata.mainservice.controllers

import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*
import tv.comnata.mainservice.services.RoomService

@RestController
class MainController(
    @Autowired private val roomService: RoomService,
) {
    @GetMapping("/isRoomExist")
    fun checkIsRoomExist(@RequestParam roomName: String): Boolean {
        return roomService.checkIsRoomExist(roomName)
    }

    @RequestMapping("/createVideo", method = [RequestMethod.PUT])
    fun createVideo(@RequestParam videoUuid: String) {
        roomService.createRoom(videoUuid)
    }

    @PostMapping("/setVideoProgress")
    fun setVideoProgress(@RequestParam videoUuid: String, @RequestParam videoProgress: Int) {
        // roomService.setVideoProgress(videoUuid, videoProgress)
    }

    companion object {
        private val logger = LoggerFactory.getLogger(MainController::class.java)
    }
}
